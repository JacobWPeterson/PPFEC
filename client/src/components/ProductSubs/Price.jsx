import React from 'react';
import styled from 'styled-components';

const NoSale = styled.p`
  margin-bottom: 1vh;
  font-weight: 300;
`;

const Discounted = styled.div`
  margin-bottom: 1vh;
  display: flex;
`;

const SaleReg = styled.p`
  font-weight: 300;
  text-decoration: line-through;
`;

const SalePrice = styled.p`
  font-weight: 300;
  color: red;
`;

const Price = ({ price, sale }) => {
  if (!sale) {
    return (
      <NoSale>{`$${Math.round(price)}`}</NoSale>
    );
  } else {
    return (
      <Discounted>
        <SaleReg>{`$${Math.round(price)}`}</SaleReg>
        <SalePrice>{`$${Math.round(sale)}`}</SalePrice>
      </Discounted>
    );
  }
};

export default Price;
