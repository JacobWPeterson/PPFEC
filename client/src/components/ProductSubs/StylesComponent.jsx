import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
  margin-bottom: 10px;
`;

const TextWrapper = styled.div`
  display: flex;
  flex-direction: row;
`;

const Style = styled.p`
  font-weight: 700;
  color: #424242;
`;

const SelectedStyle = styled.p`
  margin-left: 1vw;
  font-weight: 300;
  color: #424242;
`;

const Thumbs = styled.div`
flex-basis: 90%
display: flex;
  flex-wrap: wrap;
  margin: 0;
`;

const Image = styled.img`
  flex: 1 0 22%;
  margin: 1vh 1.2vw 1vh 0;
  border: 2px solid #aeaeae;
  border-radius: 100%;
  z-index: 0;
`;

const SelectedImage = styled.img`
  flex: 1 0 22%;
  margin: 1vh 1.2vw 1vh 0;
  border: 3px solid #80ccc4;
  border-radius: 100%;
  z-index: 0;
`;

class Styles extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentStyle: null,
      style_id: null,
    };
    this.clickHandler = this.clickHandler.bind(this);
    this.passUpStyle = this.passUpStyle.bind(this);
  }

  componentDidMount() {
    this.setState({
      currentStyle: this.props.styles.results[0].name,
      style_id: this.props.styles.results[0].style_id,
    });
  }

  clickHandler(event) {
    event.preventDefault();
    this.setState({
      currentStyle: event.target.alt,
      style_id: event.target.id,
    }, () => this.passUpStyle(this.state.style_id));
  }

  passUpStyle(style) {
    this.props.getStyleID(style);
  }

  render() {
    if (this.state.currentStyle) {
      return (
        <StyleWrapper>
          <TextWrapper>
            <Style>STYLE &gt;</Style>
            <SelectedStyle>{this.state.currentStyle.toUpperCase()}</SelectedStyle>
          </TextWrapper>
          <Thumbs>
            { this.props.styles.results.map((image) => {
              if (image.style_id === this.state.style_id) {
                return <SelectedImage onClick={this.clickHandler} key={image.style_id} id={image.style_id} src={image.photos[0].thumbnail_url} alt={image.name} width="65" height="65" />;
              }
              if (image.style_id !== this.state.style_id) {
                return <Image onClick={this.clickHandler} key={image.style_id} id={image.style_id} src={image.photos[0].thumbnail_url} alt={image.name} width="65" height="65" />;
              }
              return <div>Hello</div>;
            })}
          </Thumbs>
        </StyleWrapper>
      );
    }
    return <div>Empty</div>;
  }
}

export default Styles;
