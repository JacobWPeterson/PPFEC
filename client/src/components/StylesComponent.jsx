import React from 'react';

const Styles = () => (
  <div>
    <div>Style</div>
    <div>Selected Style</div>
    <div>Style thumbnails</div>
  </div>
);

export default Styles;
