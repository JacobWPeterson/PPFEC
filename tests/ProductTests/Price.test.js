/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Price from '../../client/src/components/ProductSubs/Price.jsx';
import StylesComponent from '../../client/src/components/ProductSubs/StylesComponent.jsx';

describe('Should show the default price', () => {
  const props = {
    noSale: { price: 40.00, sale: null },
    sale: { price: 40.00, sale: 30.00 },
  };
  const PriceComponent = mount(<Price {...props} debug />);

  it('Should render price correctly', () => {
    // mount render the whole dom with styled-component
    expect(PriceComponent).toMatchSnapshot();
  });

  it('The form should contain the regular price', () => {
    expect(PriceComponent.find(<p>{`$${Math.round(props.sale.price)}`}</p>)).toBeTruthy();
  });

  it('The form should contain the sale price when present', () => {
    expect(PriceComponent.find(<p>{`$${Math.round(props.sale.sale)}`}</p>)).toBeTruthy();
  });

  // it('The form should not render a sale price when absent', () => {
  //   expect(PriceComponent.find(<p>{`$${Math.round(props.noSale.sale)}`}</p>)).toBeFalsy();
  // });
});
