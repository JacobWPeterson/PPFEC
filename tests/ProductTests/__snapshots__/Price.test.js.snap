// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should show the default price Should render price correctly 1`] = `
<Price
  debug={true}
  price={40}
  sale={30}
>
  <Price__Discounted>
    <div
      className="Price__Discounted-eho440-1 kJaIWp"
    >
      <Price__SaleReg>
        <p
          className="Price__SaleReg-eho440-2 fMsqin"
        >
          $40
        </p>
      </Price__SaleReg>
      <Price__SalePrice>
        <p
          className="Price__SalePrice-eho440-3 fwUqGd"
        >
          $30
        </p>
      </Price__SalePrice>
    </div>
  </Price__Discounted>
</Price>
`;
